src/osd/ReplicatedPG.cc

```diff
@@ -5774,7 +5774,7 @@ int ReplicatedPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
       ctx->mod_desc.mark_unrollbackable();
       ++ctx->num_write;
       {
-	if (!obs.exists || oi.is_whiteout()) {
+	if (!obs.exists || oi.is_whiteout() || !oi.is_omap()) {
```

Comment:
For this, you're right. Before change, if object don't have omap, it return 0. After this, it return -ENOENT. It change the semantics. So apply you said.